app/views/papers/_index.html.erb

```diff
@@ -52,8 +52,13 @@
         <% end %>
       </small>
       <ul class="nav nav-pills" role="tablist">
+        <% comments_count = paper_reference ? paper_reference.comments.map {|c| c.self_and_descendants.length}.inject(0, :+).to_s : paper.references.map { |r| r.comments.map {|c| c.self_and_descendants.length}.inject(0, :+)}.inject(0, :+).to_s %>
```

Comment:
RE: Switching based on `paper_reference`, but then providing an alternate: The only reason I switch based on paper reference is to change the display depending on whether it's a on a list or not. It's only a positive - that is, if `paper_reference` exists, I display things. If you display things anyway, you might as well just remove the check for `paper_reference`. `paper.references.map...` will work just as well for both cases.